docs/snippets/ov_infer_request.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+// ! [ov:include]
+#include <openvino/openvino.hpp>
+// ! [ov:include]
+
+int main() {
+ov::Core core;
+std::shared_ptr<ov::Model> model;
+ov::CompiledModel compiled_model;
+
+//! [create_infer_request]
+auto infer_request = compiled_model.create_infer_request();
+//! [create_infer_request]
+
+//! [sync_infer]
+infer_request.infer();
+//! [sync_infer]
+
+//! [async_infer]
+infer_request.start_async();
+//! [async_infer]
+
+//! [wait]
+infer_request.wait();
+//! [wait]
+
+//! [wait_for]
+infer_request.wait_for(std::chrono::milliseconds(10));
+//! [wait_for]
+
+//! [set_callback]
+infer_request.set_callback([&](std::exception_ptr ex_ptr) { 
+    if (!ex_ptr) {
+        // all done. Output data can be processed.
+        // You can fill the input data and run inference one more time:
+        infer_request.start_async();
+    } else {
+        // Something wrong, you can analyze exception_ptr
+    }
+});
+//! [set_callback]
+
+//! [cancel]
+infer_request.cancel();
+//! [cancel]
+
+{
+//! [get_set_one_tensor]
+auto input_tensor = infer_request.get_input_tensor();
+auto output_tensor = infer_request.get_output_tensor();
+//! [get_set_one_tensor]
+}
+
+{
+//! [get_set_index_tensor]
+auto input_tensor = infer_request.get_input_tensor(0);
+auto output_tensor = infer_request.get_output_tensor(1);
+//! [get_set_index_tensor]
+}
+
+//! [get_set_tensor]
+auto tensor1 = infer_request.get_tensor("tensor_name1");
+ov::Tensor tensor2;
+infer_request.set_tensor("tensor_name2", tensor2);
+//! [get_set_tensor]
+
+{
+//! [get_set_tensor_by_port]
+auto input_port = model->input(0);
+auto output_port = model->output("tensor_name");
+ov::Tensor input_tensor;
+infer_request.set_tensor(input_port, input_tensor);
+auto output_tensor = infer_request.get_tensor(output_port);
+//! [get_set_tensor_by_port]
+}
+
+auto infer_request1 = compiled_model.create_infer_request();
+auto infer_request2 = compiled_model.create_infer_request();
+
+//! [cascade_models]
+auto output = infer_request1.get_output_tensor(0);
+infer_request2.set_input_tensor(0, output);
+//! [cascade_models]
+
+//! [roi_tensor]
+/** input_tensor points to input of a previous network and
+    cropROI contains coordinates of output bounding box **/
+ov::Tensor input_tensor;
+ov::Coordinate begin;
+ov::Coordinate end;
```

Comment:
let's have full example for 4d tensor where we want to create ROI for H and W dimensions.